test/hardhat/AqueductV1Router.spec.ts

```diff
@@ -371,4 +380,154 @@ describe("AqueductV1Router", () => {
             s
         );
     });
+
+    describe("swapExactTokensForTokens", () => {
+        const token0Amount = expandTo18Decimals(5);
+        const token1Amount = expandTo18Decimals(10);
+        const swapAmount = expandTo18Decimals(1);
+        const expectedOutputAmount = BigNumber.from("1662497915624478906");
+
+        it("happy path", async () => {
+            const { router02, token0, token1, wallet, pair } = await loadFixture(v2Fixture);
+
+            // before each
+            await token0.transfer({ receiver: pair.address, amount: token0Amount }).exec(wallet);
+            await token1.transfer({ receiver: pair.address, amount: token1Amount }).exec(wallet);
+            await pair.mint(wallet.address);
+
+            await token0.approve({ receiver: router02.address, amount: ethers.constants.MaxUint256 }).exec(wallet);
+
+            await expect(
+                router02.swapExactTokensForTokens(
+                    swapAmount,
+                    0,
+                    [token0.address, token1.address],
+                    wallet.address,
+                    ethers.constants.MaxUint256
+                )
+            )
+                .to.emit(new ethers.Contract(token0.address, erc20Abi, owner), "Transfer")
+                .withArgs(wallet.address, pair.address, swapAmount)
+                .to.emit(new ethers.Contract(token1.address, erc20Abi, owner), "Transfer")
+                .withArgs(pair.address, wallet.address, expectedOutputAmount)
+                .to.emit(pair, "Sync")
+                .withArgs(token0Amount.add(swapAmount), token1Amount.sub(expectedOutputAmount))
+                .to.emit(pair, "Swap")
+                .withArgs(router02.address, swapAmount, 0, 0, expectedOutputAmount, wallet.address);
+        });
+
+        it("amounts", async () => {
+            const { router02, token0, token1, wallet, pair, RouterEmit, factoryV2 } = await loadFixture(v2Fixture);
+
+            // need to allow RouterEmit to call swap() (it's using router02.delegateCall())
+            await factoryV2.setAuction(RouterEmit.address);
```

Comment:
1. What's `RouterEventEmitter` needed for exactly?
2. Can you explain why this line is needed compared to the original code?

Reply:
1. RouterEventEmitter and its associated tests are taken directly from Uniswap V2 peripheral contracts. It seems this is used to test that the return values of the calls to the router can be properly emitted as events
2. factory.setAuction() is used to control the interface to pool.swap() (ie pool.swap() can only be called by whatever address is set with factory.setAuction()). In order to use the auction, we deploy that contract and call setAuction() with its address. To start out, we 'turn off' the auction by calling setAuction with the address of the router.